droidlet/memory/sql_memory.py

```diff
@@ -1174,4 +1174,5 @@ def register_hook(self, receiver, sender):
         """
         allows for registering hooks using the event dispatcher
         """
-        self.dispatch_signal.connect(receiver, sender)
+        if sender == self.db_write:
+            self.dispatch_signal.connect(receiver, sender)
```

Comment:
the else has to be an error.

Reply:
Made an else case in [0b79202](https://github.com/facebookresearch/droidlet/pull/500/commits/0b79202252af9a452605bda27c4eeb61a30c256d) to raise a ValueError for both cases you pointed out. Let me know how that looks!